apps/frontend/src/components/global/groups/GroupModal.vue

```diff
@@ -240,11 +240,17 @@ export default class GroupModal extends Vue {
         updateGroupUserRole
       );
     });
-    return Promise.all([
-      ...addedUserPromises,
-      ...removedUserPromises,
-      ...updatedUserPromises
-    ]);
+    const removedUserPromises = toRemove.map((user) => {
+      const removeUserDto: IRemoveUserFromGroup = {
+        userId: user.id
+      };
+      return axios.delete(`/groups/${group.id}/user`, {data: removeUserDto});
+    });
+    return Promise.all([...addedUserPromises, ...updatedUserPromises]).then(
+      () => {
+        return removedUserPromises;
```

Comment:
removedUserPromises is an array of promises.  this function being wrapped in a 'then' means that the function's return value is immediately a promise, so doing an await against it will return the array of promises that themselves haven't been awaited, i.e. line 191 isn't doing anything worthwhile.  this line should probably be `return Promise.all(removedUserPromises);`

async is hard to do right.  make sure to walk through the logic every time.